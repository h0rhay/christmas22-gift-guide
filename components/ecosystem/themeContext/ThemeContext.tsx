import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
