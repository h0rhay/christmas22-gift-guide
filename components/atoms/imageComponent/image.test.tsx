import { render } from "@testing-library/react";
import ImageComponent from ".";
import { ThemeProvider } from "styled-components";
import Theme from "styles/theme";
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

const renderComponent = () =>
  render(
    <ThemeProvider theme={Theme}>
      <ThemeContext colorGroup="hero">
        <ImageComponent
          className=""
          type="product"
          altText="product"
          src={"/img/authors/marc_jacobs_test.jpg"}
        />
      </ThemeContext>
    </ThemeProvider>
  );

describe("Image Component", () => {
  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
