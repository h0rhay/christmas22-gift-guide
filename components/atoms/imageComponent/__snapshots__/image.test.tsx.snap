// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="imageComponent__StyledImage-sc-y6629v-0 BNymA image-component__container"
      type="product"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt="product"
          class=""
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(/img/authors/marc_jacobs_test.jpg); background-position: 0% 0%;"
        />
        <noscript />
      </div>
    </div>
  </div>
</div>
`;
